RA-28-Jetpack-Navigation

DenisGubernatorov opened the pull request:
1) Повышение targetJdk до 35 версии
2) Реализация навигации между фрагментами с использованием jetpack navigation.

DenisGubernatorov committed:
RA-28-Jetpack-Navigation

1) Повышение targetSdk

DenisGubernatorov committed:
RA-28-Jetpack-Navigation

1) Реализация навигации между фрагментами с использованием jetpack navigation

DenisGubernatorov commented:
navigationFragment = "2.5.3"
navigationFragmentKtx = "2.5.3"
navigationUiKtx = "2.5.3" 
Пришлось даунгрейдить до 2.5.3. С версиями выше приложение крашилось при переходе на список с рецептами. Типа это известный баг в google

aGroupoid commented:
Лучше тогда проапгрейдить другие либы до максимальной версии, чтоб конфликт решить

aGroupoid commented on app/src/main/java/com/example/recipesapp/ui/MainActivity.kt:
```diff
@@ -20,27 +16,14 @@ class MainActivity : AppCompatActivity() {
         binding.category.setOnClickListener {
-            loadFragment<CategoriesListFragment>()
-
+            findNavController(R.id.mainContainer).navigate(R.id.recipesListFragment)
```
Одна из проблем прошлой реализации и она сейчас присутсвует, но на jetpack navigetion мы её можем поправить:
Попробуй 10 раз подряд нажать на кнопку Категорий, все они попадут в backstack, чтобы их все закрыть придется нажать 10 раз на кнопку назад.

Добавь условие if (navController.currentDestination?.id != R.id.categoriesListFragment), которое проверяет что текущий фрагмент уже не запущен

aGroupoid commented on app/src/main/java/com/example/recipesapp/ui/MainActivity.kt:
```diff
@@ -20,27 +16,14 @@ class MainActivity : AppCompatActivity() {
 
         binding.favorites.setOnClickListener {
-            loadFragment<FavoritesFragment>()
+            findNavController(R.id.mainContainer).navigate(R.id.favoritesFragment)
```
Добавь глобальные экшены для этих мест

DenisGubernatorov commented:
> Лучше тогда проапгрейдить другие либы до максимальной версии, чтоб конфликт решить

Не очень понятно что проапгрейдить. Судя по тому, что идешка не подчеркивает ничего,кроме того, про что написал,  все последних версий 
![image](https://github.com/user-attachments/assets/fdc7b04e-1330-426e-87c0-63f52d763466)

DenisGubernatorov committed:
RA-28-Jetpack-Navigation

1) Экшены перехода во фрагменты избранного и списка рецепта сделаны глобальными
2) Добавлена проверка текущего фрагмента.Что бы избежать множественного помещения списка категорий в backstack

DenisGubernatorov committed:
RA-28-Jetpack-Navigation

1) Исправлена иницализация navcontroller

aGroupoid reviewed (approved):
Всё верно, апрув. По версиям - можешь оставить так, но как минимум версия Котлин уже 2.1.0+ вышла, у тебя - 1.9.0. Если интересно, почитай в интернете как мигрировать на актуальную версию котлина
